### PR TITLE
[Mobile Payments] Add ChargeID to Order

### PR DESCRIPTION
**Pull request by joshheald**
Part of: #5976 

### PR Length
Please don't be too worried about the length of this PR - the majority of the changes are test data (+183) or changing to use `Order.fake()` instead of `Order()` when building test data (+207, -416). This will mean that future changes to `Order` don't require large numbers of changes in other test files.

When changing to use `Order.fake().copy()` to change the value of the sample data, I have removed the unnecessary parameters from the call which would have explicitly set a property to the default empty value that `.fake()` sets in the first place.

### Description

We need the `chargeID` to be able to fetch `/payments/charges/<charge_id>`, which will give us various details about a payment that's been taken, in particular the last 4 digits of the card, and the card type.

This adds the `chargeID` to `Networking.Order`, and adds `meta_data` to the `_fields` when loading a single order so that we receive the `charge_id`.

Further PRs will add the ability to fetch a `Charge`, which we will use when displaying payment details in the UI. This will also be beneficial for fetching details for printing receipts, avoiding the need to store that locally on device and the limitation that IPP receipts must be printed from the same device as the payment was taken from.

### Testing instructions

Consider only testing as a part of the further changes where we will fetch the details of charges for showing the last 4 digits of a card number in the order details.

You can test the decoding by running the app from Xcode, with a breakpoint in [`Order.init(from:)`](https://github.com/woocommerce/woocommerce-ios/blob/a4c10b8982207f0542f5e1477d451b58b968a5b1/Networking/Networking/Model/Order.swift#L148) after the `chargeID` is decoded.

1. Enable the simulated card reader
2. Launch the app and go to `Orders`
3. Tap + and take an IPP for the order you add
4. Return to the `Orders` root screen, and enable your breakpoint
5. Tap the order, and you should hit the breakpoint and be able to see that a `chargeID` is decoded. Typically these look like: `ch_3KMtak2EdyGr1FMV02G9Qqq1` 

### Screenshots

N/A no UI changes

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Yosemite/Yosemite/Model/Storage/Order+ReadOnlyConvertible.swift [hunk 1]
```diff
@@ -71,7 +71,7 @@ extension Storage.Order: ReadOnlyConvertible {
         let orderShippingLines = shippingLines?.map { $0.toReadOnly() } ?? [Yosemite.ShippingLine]()
         let orderFeeLines = fees?.map { $0.toReadOnly() } ?? [Yosemite.OrderFeeLine]()
         let orderTaxLines = taxes?.map { $0.toReadOnly() } ?? [Yosemite.OrderTaxLine]()
-
+        #warning("ChargeID needs to be set here")
```
Comment (joshheald) [inline Yosemite/Yosemite/Model/Storage/Order+ReadOnlyConvertible.swift:74]: I will remove this warning when I update the Storage/Yosemite layers, which is a fast follow and so I won't merge this warning to trunk.

File: Networking/Networking/Model/Order.swift [hunk 2]
```diff
@@ -140,6 +143,11 @@ public struct Order: Decodable, GeneratedCopiable, GeneratedFakeable {
         let paymentMethodID = try container.decode(String.self, forKey: .paymentMethodID)
         let paymentMethodTitle = try container.decode(String.self, forKey: .paymentMethodTitle)
 
+        var chargeID: String? = nil
+        if let allOrderMetaData = try? container.decode([OrderMetaData].self, forKey: .metadata) {
```
Comment (koke) [inline Networking/Networking/Model/Order.swift]: Probably a pretty minor performance improvement, but we seem to be decoding metadata again a few lines down for the shipping address. Perhaps we can share the optional `allOrderMetaData` for both cases?

File: Networking/Networking/Remote/OrdersRemote.swift [hunk 3]
```diff
@@ -272,7 +272,7 @@ public extension OrdersRemote {
             "billing", "coupon_lines", "shipping_lines", "refunds", "fee_lines", "order_key", "tax_lines"
         ]
         private static let singleOrderExtraFieldValues = [
-            "line_items", "shipping"
+            "line_items", "shipping", "meta_data"
```
Comment (koke) [inline Networking/Networking/Remote/OrdersRemote.swift:275]: If we have to request this field explicitly, now I'm wondering if the existing usage of extracting the shipping address from the metadata wasn't working, or am I missing something else?

File: Yosemite/Yosemite/Model/Storage/Order+ReadOnlyConvertible.swift (hunk 1 above)
Comment (koke) [reply Yosemite/Yosemite/Model/Storage/Order+ReadOnlyConvertible.swift:74]: 👍🏽 to cross-reference: https://github.com/woocommerce/woocommerce-ios/pull/6019/files#diff-2c5b8cf4b92fce8b6c74654d9561628dfcda87854d4cd9b77a8bb31554f63bafL74

**Review by koke — APPROVED** [review]
I didn't follow the exact testing steps. Instead, I just pulled to refresh the orders list (most everything there has been paid through the app) and the API response doesn't seem to have any metadata. 

However, when I navigate to a specific order and it requests the order data, I can see the charge ID.

**Comment by joshheald** [comment]
> I didn't follow the exact testing steps. Instead, I just pulled to refresh the orders list (most everything there has been paid through the app) and the API response doesn't seem to have any metadata.
> 
> However, when I navigate to a specific order and it requests the order data, I can see the charge ID.

Yeah, that was deliberate because I figured we didn't actually need the metadata/chargeID in the list, so I only added it to `singleOrderExtraFieldValues` in the OrdersRemote. Do you think there's any issue with that which I've not seen?

File: Networking/Networking/Model/Order.swift (hunk 2 above)
Comment (joshheald) [reply Networking/Networking/Model/Order.swift]: Yeah I'll make that change just now

File: Networking/Networking/Remote/OrdersRemote.swift (hunk 3 above)
Comment (joshheald) [reply Networking/Networking/Remote/OrdersRemote.swift:275]: No, I don't think it was working either. It was only the `shipping_phone` which was extracted, and only for WooCommerce <5.6.0 according to the comment

**Comment by koke** [comment]
> Do you think there's any issue with that which I've not seen?

I think it's OK, I just wasn't aware we were loading a different set of fields for single vs multiple requests. I guess it helps with performance, and we're already reloading everything when you navigate to an order. I was a bit conflicted about this, but it's more about the general networking in the app than anything specific to this PR

**Pull request merged**